src/nyc_trees/apps/event/views.py

```diff
@@ -52,9 +53,22 @@ def _process_event_form(form, request):
     except ValueError:
         # Lat/lng were not submitted or could not be converted to floats.
         pass
+
+    if event:
+        event_location = event.location
+
     is_valid = form.is_valid()
+
     if is_valid:
-        form.save()
+        needs_pdf_map = (
+            event is None or
+            not event_location.equals_exact(form.data['location']))
```

Comment:
We don't need to because changing the slug doesn't change the map, and the PDF path is already stored. No harm in returning a PDF whose filename contains an old slug.